src/generic/AbsMSeries.jl

```diff
@@ -716,37 +716,36 @@ end
 function power_series_ring(R::AbstractAlgebra.Ring, prec::Vector{Int},
                   s::Vector{Symbol}; cached::Bool=true, model=:capped_absolute)
     U = elem_type(R)
- 
+
     S, _ = AbstractAlgebra.polynomial_ring(R, s)
     V = elem_type(S)
 
-    if model == :capped_absolute
-       parent_obj = AbsMSeriesRing{U, V}(S, prec, s, cached)
-    else
-       error("Unknown model")
-    end
- 
-    return tuple(parent_obj, gens(parent_obj))
+    model === :capped_absolute || error("Unknown model")
+
+    parent_obj = AbsMSeriesRing{U, V}(S, prec, s, cached)
+
+    return parent_obj, gens(parent_obj)
 end
 
-function power_series_ring(R::AbstractAlgebra.Ring, weights::Vector{Int}, prec::Int,
-   s::Vector{Symbol}; cached::Bool=true, model=:capped_absolute)
-   U = elem_type(R)
+function power_series_ring(R::AbstractAlgebra.Ring, prec::Int,
+        s::Vector{Symbol}; weights::Union{Vector{Int}, Nothing}=nothing,
+        cached::Bool=true, model=:capped_absolute)
+    U = elem_type(R)
 
-   S, _ = AbstractAlgebra.polynomial_ring(R, s)
-   V = elem_type(S)
+    S, _ = AbstractAlgebra.polynomial_ring(R, s)
+    V = elem_type(S)
 
-   if model == :capped_absolute
-      parent_obj = AbsMSeriesRing{U, V}(S, weights, prec, s, cached)
-   else
-      error("Unknown model")
-   end
+    model === :capped_absolute || error("Unknown model")
 
-   return tuple(parent_obj, gens(parent_obj))
-end
+    if weights === nothing
+        parent_obj = AbsMSeriesRing{U, V}(S, [prec for _ in s], s, cached)
+    else
+        parent_obj = AbsMSeriesRing{U, V}(S, weights, prec, s, cached)
+    end
 
-function power_series_ring(R::AbstractAlgebra.Ring, prec::Int,
-                  s::Vector{Symbol}; cached::Bool=true, model=:capped_absolute)
-    prec_vec = [prec for v in s]
-    return power_series_ring(R, prec_vec, s; cached=cached, model=model)
+    return parent_obj, gens(parent_obj)
 end
+
+power_series_ring(R::AbstractAlgebra.Ring, weights::Vector{Int},
```

Comment:
Maybe add a comment:
```suggestion
# for backwards compatibility
power_series_ring(R::AbstractAlgebra.Ring, weights::Vector{Int},
```
And maybe this should be moved to `src/Deprecations.jl` ?

Is that OK for you, @thofma and @fieker ?